### PR TITLE
Fix shift selection after pan

### PR DESCRIPTION
**Pull request by etpinard**
Fixes https://github.com/plotly/plotly.js/issues/2669

before:

peek 2018-05-28 17-51

now:

peek 2018-05-28 17-52

cc @alexcjohnson

### CODE REVIEW

File: src/plots/cartesian/dragbox.js [hunk 1]
```diff
@@ -178,6 +178,8 @@ function makeDragBox(gd, plotinfo, x, y, w, h, ns, ew) {
             prepSelect(e, startX, startY, dragOptions, dragModeNow);
         } else {
             dragOptions.clickFn = clickFn;
+            // clear selection polygon cache (if any)
+            plotinfo.selection = false;
```
Comment (alexcjohnson) [inline src/plots/cartesian/dragbox.js]: Should this be part of `clearSelect`? That gets called in all 3 blocks below (directly in the first and third, and in the second it's inside `zoomPrep` - looks like these can be simplified anyway) but also in [`zoomWheel`](https://github.com/plotly/plotly.js/blob/7784a10880545ef79b5dbc269d5450f337c6807d/src/plots/cartesian/dragbox.js#L397) - does that case have the same bug?

Comment (etpinard) [reply src/plots/cartesian/dragbox.js]: Ah. Good catch. Zoom has the problem. 

Working on a fix :pick:

File: src/plots/cartesian/dragbox.js [hunk 2]
```diff
@@ -393,9 +393,7 @@ function makeDragBox(gd, plotinfo, x, y, w, h, ns, ew) {
             return;
         }
 
-        if(redrawTimer === null) {
-            clearSelect(zoomlayer);
-        }
+        clearSelect(zoomlayer);
```
Comment (etpinard) [inline src/plots/cartesian/dragbox.js]: this fixes what I botched in https://github.com/plotly/plotly.js/commit/51c3128718ff61ccaad259a7ff65b76c06b2d2a3

Comment (etpinard) [reply src/plots/cartesian/dragbox.js]: Oh wait, this worked fine in https://github.com/plotly/plotly.js/commit/51c3128718ff61ccaad259a7ff65b76c06b2d2a3

It is the optimization of https://github.com/plotly/plotly.js/commit/bb022818e7fe6c2c808ad440a633558398853325 that broke this (except during the very first scroll event),

File: src/plots/cartesian/dragbox.js (hunk 1 above)
Comment (etpinard) [reply src/plots/cartesian/dragbox.js]: fixed and :lock: in https://github.com/plotly/plotly.js/pull/2676/commits/f682fb8979da56083c4a513fe7cbe642f323a2e3

File: src/plots/cartesian/dragbox.js [hunk 3]
```diff
@@ -196,11 +194,18 @@ function makeDragBox(gd, plotinfo, x, y, w, h, ns, ew) {
             } else if(dragModeNow === 'pan') {
                 dragOptions.moveFn = plotDrag;
                 dragOptions.doneFn = dragTail;
-                clearSelect(zoomlayer);
+                clearAndResetSelect();
```
[the remark this answers is not in the document]
Comment (etpinard) [reply src/plots/cartesian/dragbox.js]: Ooops, I missed the one in `zoomPrep`. Cleaned up in -> https://github.com/plotly/plotly.js/pull/2676/commits/34a0cf7faefa75caffd3c572b80209f9ddc6303d

**Pull request merged**